p2p/transport_test.go

```diff
@@ -138,46 +138,50 @@ func TestTransportMultiplexConnFilterTimeout(t *testing.T) {
 }
 
 func TestTransportMultiplexMaxIncomingConnections(t *testing.T) {
+	pv := ed25519.GenPrivKey()
+	id := PubKeyToID(pv.PubKey())
 	mt := newMultiplexTransport(
-		emptyNodeInfo(),
+		testNodeInfo(
+			id, "transport",
+		),
 		NodeKey{
-			PrivKey: ed25519.GenPrivKey(),
+			PrivKey: pv,
 		},
 	)
-	id := mt.nodeKey.ID()
 
 	MultiplexTransportMaxIncomingConnections(0)(mt)
 
 	addr, err := NewNetAddressString(IDAddressString(id, "127.0.0.1:0"))
 	if err != nil {
 		t.Fatal(err)
 	}
-
+	maxInbound := 2 // Setting max to 0 actually means unlimited incoming connections
+	MultiplexTransportMaxIncomingConnections(maxInbound)(mt)
 	if err := mt.Listen(*addr); err != nil {
 		t.Fatal(err)
 	}
 
-	errc := make(chan error)
+	laddr := NewNetAddress(mt.nodeKey.ID(), mt.listener.Addr())
 
-	go func() {
-		addr := NewNetAddress(id, mt.listener.Addr())
+	// Connect more peers than max
+	for i := 0; i <= maxInbound; i++ {
+		errc := make(chan error)
+		go testDialer(*laddr, errc)
 
-		_, err := addr.Dial()
-		if err != nil {
-			errc <- err
-			return
+		if i < maxInbound {
+			if err := <-errc; err != nil {
```

Comment:
looks like err := <-errc could be extracted from both branches

Reply:
Nice one - done